File: Data Structures & Algorithms/reverse-linked-list-ii/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:

    def reverse(self, node, length):
        prev, start = None, node
        while node and length:
            next_node = node.next
            node.next = prev
            prev = node
            node = next_node 
            length -= 1
        start.next = node
        return prev


    def reverseBetween(self, head: Optional[ListNode], left: int, right: int) -> Optional[ListNode]:
        length = right - left + 1
        if left == 1:
            return self.reverse(head, length)
        
        curr = head
        i = 1
        while curr and i + 1 != left:
            i += 1
            curr = curr.next
        
        curr.next = self.reverse(curr.next, length)
        return head

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int left = 1;
        for (int right = 1; right < nums.size(); right++) {
            int prev = nums[right - 1];
            if (nums[right] != prev) {
                nums[left++] = nums[right];
            } 

        }

        return left;
    }
};